### PR TITLE
systemd: mount the squashfs with nodev

### PR DESCRIPTION
**Pull request by mvo5**

### CODE REVIEW

File: systemd/systemd.go [hunk 1]
```diff
@@ -437,12 +437,12 @@ func MountUnitPath(baseDir string) string {
 }
 
 func (s *systemd) WriteMountUnitFile(name, what, where, fstype string) (string, error) {
-	extra := ""
+	options := []string{"nodev"}
```
Comment (zyga) [inline systemd/systemd.go]: I think you want this unless core snap

Comment (pedronis) [inline systemd/systemd.go]: on classic we mount /dev from the real root on top of the core snap anyway, no?
on core the mounting of core on / is not done by this unit I think?

so do we really need to have real devices under /snap/core/current/dev ?

Comment (pedronis) [inline systemd/systemd.go]: the basic template btw has this:

/dev/ r,
/dev/**/ r,

so we are probably confused about something

either way we can probably use /dev/{u,}random to write a test that access to that still works

Comment (zyga) [inline systemd/systemd.go]: @pedronis devices in cgroup is totally unrelated, it only works when tagging is used. On core I don't know how this happens but we mount core from the initrd. On classic we overmount /dev so maybe this PR can go in as-is.

Comment (niemeyer) [inline systemd/systemd.go]: @zyga We have core tests in our spread suite, and apparently they pass?

**Comment by niemeyer** [comment]
Tests pass, the change is straightforward, and we all agree this is important. So I'll go ahead and merge this.

File: systemd/systemd.go (hunk 1 above)
Comment (ogra1) [inline systemd/systemd.go]: we use devtmpfs on core which gets mounted by the initrd so the core snap itself never needs to manage /dev ... not sure if we might hit issues when the rootfs itself is mounted with nodev but i doubt it ... (we'll see in edge once this landed and i'll complain if all my boards stop working ;) ... i run everything on edge here)

**Pull request merged**